shop/src/main/java/ro/msg/learning/service/impl/RevenueServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+package ro.msg.learning.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import ro.msg.learning.entity.Location;
+import ro.msg.learning.entity.Revenue;
+import ro.msg.learning.repository.RevenueRepository;
+import ro.msg.learning.repository.RevenueRepository.UnprocessedRevenueInfo;
+import ro.msg.learning.service.interfaces.LocationService;
+import ro.msg.learning.service.interfaces.RevenueService;
+
+@Service
+@RequiredArgsConstructor
+public class RevenueServiceImpl implements RevenueService {
+	private static final String SQL_PLACEHOLDER = "%";
+	
+	private final RevenueRepository revenueRepository;
+	private final LocationService locationService;
+
+	@Override
+	public List<Revenue> getRevenuesForDate(final LocalDate date) {
+		return revenueRepository.getRevenuesForDate(getDateAsSqlString(date));
+	}
+
+	@Override
+	public void processTodaysRevenue() {
+		final LocalDate todaysDate = LocalDate.now();
+
+		final List<UnprocessedRevenueInfo> unprocessedRevenueData = revenueRepository
+				.getRevenueInfoForDate(getDateAsSqlString(LocalDate.now()));
+
+		final Map<Integer, List<UnprocessedRevenueInfo>> revenueDataGroupedByLocation = getRevenueDataGroupedByLocation(
+				unprocessedRevenueData);
+
+		final Map<Integer, BigDecimal> todaysProfitsPerLocation = computeTodaysProfitsForEachLocation(
+				revenueDataGroupedByLocation);
+
+		persistTodaysRevenues(todaysDate, todaysProfitsPerLocation);
+	}
+
+	private String getDateAsSqlString(final LocalDate date) {
+		final StringBuilder formattedDate = new StringBuilder();
```

Comment:
you could implement a Converter from/to LocalDate to/from sql.Date extending AttributeConverter<>